### PR TITLE
Add miktex package

### PR DESCRIPTION
**Pull request by vmiheer**
Tested with following these [postinstall](https://miktex.org/howto/build-unx) instructions and compiling `pdflatex sample2e`.

### CODE REVIEW

**Comment by vmiheer** [comment]
@alalazo, @tgamblin I was wondering if you could review this pull request or please add appropriate reviewer? Thank you :)

**Comment by vmiheer** [comment]
Adding George who is original author of texlive package.

**Comment by vmiheer** [comment]
Gentle reminder, @tgamblin, @alalazo, @hartzell :)

**Comment by vmiheer** [comment]
Ping!

File: var/spack/repos/builtin/packages/libmspack/package.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+class Libmspack(AutotoolsPackage):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libmspack/package.py]:
```suggestion
# SPDX-License-Identifier: (Apache-2.0 OR MIT)

from spack.package import *


class Libmspack(AutotoolsPackage):
```

File: var/spack/repos/builtin/packages/miktex/package.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/miktex/package.py]:
```suggestion
from spack.package import *
```

File: var/spack/repos/builtin/packages/libmspack/package.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    """libmspack is a portable library for some loosely related Microsoft
+    compression formats"""
+    maintainers = ['vmiheer']
+    homepage = "https://www.cabextract.org.uk/libmspack/"
+    url = "https://www.cabextract.org.uk/libmspack/libmspack-0.10.1alpha.tar.gz"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libmspack/package.py]: Why not use the git repository at https://github.com/kyz/libmspack/?  

Note the sha256 is different for the same version on GitHub (`bd8537b04ebd2149819ce867e3eba50f862a8a0043fd2061360b4101eddaa382`).

File: var/spack/repos/builtin/packages/miktex/package.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+class Miktex(CMakePackage):
+    """MiKTeX is a modern C/C++ implementation of TeX & Friends for Windows,
+    macOS and Linux."""
+    maintainers = ['vmiheer']
+    homepage = "https://miktex.org/"
+    url = "https://miktex.org/download/ctan/systems/win32/miktex/source/miktex-22.1.tar.xz"
+
+    version('22.1', sha256='f77caf932e3405d42365438f5c735fe4765989c3442fbb0e9114ad56e7515423',
+            url='https://miktex.org/download/ctan/systems/win32/miktex/source/miktex-22.1.tar.xz')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/miktex/package.py]: This argument duplicates the `url` at line 14 so is not needed.

File: var/spack/repos/builtin/packages/miktex/package.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+from spack import *
+
+
+class Miktex(CMakePackage):
+    """MiKTeX is a modern C/C++ implementation of TeX & Friends for Windows,
+    macOS and Linux."""
+    maintainers = ['vmiheer']
+    homepage = "https://miktex.org/"
+    url = "https://miktex.org/download/ctan/systems/win32/miktex/source/miktex-22.1.tar.xz"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/miktex/package.py]: Why not use the GitHub version (`https://github.com/MiKTeX/miktex/archive/refs/tags/22.1.tar.gz`)?

The sha256 is different (`a9013a424a5a70ddab08c6b8f1693a22d5da9081399228c37a80a2e733642df2`).

File: var/spack/repos/builtin/packages/miktex/package.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+    version('22.1', sha256='f77caf932e3405d42365438f5c735fe4765989c3442fbb0e9114ad56e7515423',
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/miktex/package.py]: There are reported bug fixes in version `22.3` on `GitHub`.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
There are `isort` issues that have to be resolved before the PR can be merged.  I also have a few questions.

In addition, it is easier to review PRs if you create a PR-specific (i.e., non-`develop`) branch for them.

**Pull request closed without merging**